src/Stratis.Bitcoin.Features.Wallet/Wallet.cs

```diff
@@ -728,12 +728,29 @@ public IEnumerable<UnspentOutputReference> GetSpendableTransactions(int currentC
 
                     if (confirmationCount >= confirmations)
                     {
-                        yield return new UnspentOutputReference
+                        if (includeImmature || (!(transactionData.IsCoinBase ?? false) && !(transactionData.IsCoinStake ?? false)))
                         {
-                            Account = this,
-                            Address = address,
-                            Transaction = transactionData
-                        };
+                            // This output can unconditionally be included in the results.
+
+                            yield return new UnspentOutputReference
```

Comment:
I feel this whole new `if` can be written better so that we avoid code duplication here.